sample-schemas/demo_schema.erl

```diff
@@ -43,8 +43,16 @@ fields("x.y") ->
 
 fields("j.k") ->
     [ {"some_int", fun priv_int/1}
-    ].
+    ];
+
+fields(person) ->
+    [{id, fun(mapping) -> "person.id"; (type) -> "id"; (_) -> undefined end}];
 
+fields("id") ->
```

Comment:
how about
`-define(FIELD(NAME, TYPE), {NAME, fun(mapping) -> NAME; (type) -> TYPE; (_) -> undefined end}).`

Reply:
in emqx.schema I added helper function `ref`
https://github.com/emqx/hocon/pull/62/files#diff-c503cb08c8b7abf5ed5c983df29909f57f33a33ad139020f8b0022904fd6ef6fR1359